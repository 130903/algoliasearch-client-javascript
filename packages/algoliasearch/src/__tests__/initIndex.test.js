// @flow

import { initIndex } from '../';

const validParams = {
  appId: 'some_app',
  apiKey: 'some_key',
  indexName: 'some_index',
};

const snapshotAll = requests =>
  requests.map(req => req.then(sn => expect(sn).toMatchSnapshot()));

it('throws when it has too little parameters', () => {
  // $FlowIssue --> type disallows this
  expect(() => initIndex({})).toThrow();
  // $FlowIssue --> type disallows this
  expect(() => initIndex({ appId: '' })).toThrow();
  // $FlowIssue --> type disallows this
  expect(() => initIndex({ apiKey: '' })).toThrow();
  // $FlowIssue --> type disallows this
  expect(() => initIndex({ apiKey: '', appId: '' })).toThrow();

  expect(() => initIndex(validParams)).not.toThrow();
});

it('contains the correct methods', () => {
  const index = initIndex(validParams);
  expect(Object.keys(index)).toMatchSnapshot();
});

it('batch', () => {
  const index = initIndex(validParams);
  const requests = [
    index.batch([
      {
        action: 'addObject',
        body: {
          some: 'data',
          nested: { is: { all: 'fine' } },
          function: is => `${is}also allowed, but not smart`,
        },
      },
    ]),
    index.batch([
      {
        action: 'updateObject',
        body: {
          objectID: 'some-cool-object',
          some: 'other data',
          and: 'this is everything now',
        },
      },
    ]),
    index.batch([
      {
        action: 'partialUpdateObject',
        body: {
          objectID: 'some-cool-object',
          some: 'data',
          the: 'rest is still here',
        },
      },
    ]),
    index.batch([
      {
        action: 'partialUpdateObjectNoCreate',
        body: {
          objectID: 'some-cool-object',
          this: 'object',
          "won't": 'be made',
          if: "it doesn't exist yet",
        },
      },
    ]),
    index.batch([
      {
        action: 'deleteObject',
        body: {
          objectID: 'some-uncool-object',
        },
      },
    ]),
    index.batch([
      {
        action: 'delete',
      },
    ]),
    index.batch([
      {
        action: 'clear',
      },
    ]),
  ];

  snapshotAll(requests);
});

it('clear', () => {
  const index = initIndex(validParams);

  const requests = [index.clear()];
  snapshotAll(requests);
});

it('copy', () => {
  const index = initIndex(validParams);

  const requests = [index.copy('new_index')];
  snapshotAll(requests);
});

it('remove', () => {
  const index = initIndex(validParams);

  const requests = [index.remove()];
  snapshotAll(requests);
});

it('move', () => {
  const index = initIndex(validParams);

  const requests = [index.move('new_index')];
  snapshotAll(requests);
});

it('browse', () => {
  const index = initIndex(validParams);

  const requests = [index.browse(), index.browse({ hitsPerPage: 40 })];

  snapshotAll(requests);
});

it('browseFrom', () => {
  const index = initIndex(validParams);

  const requests = [index.browseFrom('some_weirdo_cursor')];
  snapshotAll(requests);
});

it('search', () => {
  const index = initIndex(validParams);

  const requests = [index.search({}), index.search({ hitsPerPage: 4 })];
  snapshotAll(requests);
});

it('waitTask', () => {
  const index = initIndex(validParams);
  const requests = [index.waitTask('some_task_id')];

  snapshotAll(requests);
});

// "waitTask",
