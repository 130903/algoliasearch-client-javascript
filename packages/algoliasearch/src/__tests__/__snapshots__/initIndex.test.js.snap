// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`batch 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "addObject",
        "body": Object {
          "function": [Function],
          "nested": Object {
            "is": Object {
              "all": "fine",
            },
          },
          "some": "data",
        },
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 2`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "updateObject",
        "body": Object {
          "and": "this is everything now",
          "objectID": "some-cool-object",
          "some": "other data",
        },
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 3`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "partialUpdateObject",
        "body": Object {
          "objectID": "some-cool-object",
          "some": "data",
          "the": "rest is still here",
        },
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 4`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "partialUpdateObjectNoCreate",
        "body": Object {
          "if": "it doesn't exist yet",
          "objectID": "some-cool-object",
          "this": "object",
          "won't": "be made",
        },
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 5`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "deleteObject",
        "body": Object {
          "objectID": "some-uncool-object",
        },
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 6`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "delete",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`batch 7`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "action": "clear",
      },
    ],
  },
  "method": "POST",
  "path": "/1/indexes//batch",
}
`;

exports[`contains the correct methods 1`] = `
Array [
  "batch",
  "clear",
  "copy",
  "remove",
  "browse",
  "browseFrom",
  "move",
  "search",
  "similarSearch",
  "waitTask",
]
`;

exports[`search 1`] = `
Object {
  "body": Object {
    "params": Array [
      Object {
        "params": Object {},
      },
    ],
  },
  "forceReadHosts": true,
  "method": "POST",
  "path": "/1/indexes//query",
}
`;
